### PR TITLE
adding tip on hoisting

### PR DESCRIPTION
**Pull request by seanpat09**

### CODE REVIEW

**Comment by manmadareddy** [comment]
:+1:

**Comment by mvedie** [comment]
:+1:

File: README.md [hunk 1]
```diff
@@ -18,6 +18,49 @@ To get updates, watch the repo and follow the [Twitter account](https://twitter
 
 # Tips list
 
+## #07 - Hoisting
+> 2016-01-07 by [@squizzleflip](https://twitter.com/squizzleflip)
+
+Understanding hoisting will help you organize your function scope. Just remember, variable declaration and function definition are hoisted to the top. Variable definition is not, even if you declare and define a variable on the same line.
```
Comment (loverajoel) [inline README.md]: [hoisting](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var#var_hoisting)

File: README.md [hunk 2]
```diff
@@ -18,6 +18,49 @@ To get updates, watch the repo and follow the [Twitter account](https://twitter
 
 # Tips list
 
+## #07 - Hoisting
+> 2016-01-07 by [@squizzleflip](https://twitter.com/squizzleflip)
```
Comment (loverajoel) [inline README.md]: update the date to today and the number tip

File: README.md [hunk 3]
```diff
@@ -18,6 +18,49 @@ To get updates, watch the repo and follow the [Twitter account](https://twitter
...
+  functionVar();
+
+  var functionVar = function(){
+    console.log('I did it!');
+  }
+}
+```
+
+To make things easier to read, declare all of your variables at the top of your function scope so it is clear which scope the variables are coming from. Define your variables before you need to use them. Define your functions at the bottom of your scope to keep them out of your way. 
+
```
Comment (loverajoel) [inline README.md:62]: maybe you can explain the difference between declaration and initialisation about vars

**Comment by loverajoel** [comment]
please, squash your commits

File: README.md (hunk 3 above)
Comment (ghengeveld) [inline README.md:62]: Declaring variables at the top only goes for var, [not for let and const](https://github.com/ghengeveld/react-redux-styleguide#variables-declaration-and-usage).

+1 on declaring functions at the bottom.

**Comment by loverajoel** [comment]
hey @seanpat09 can you make the changes today, so I can merge and publish this tip today?

**Comment by seanpat09** [comment]
@loverajoel Sorry about that, just got back today. I made the change you suggested, though my git is pretty weak so I'm not sure if I made this pull request properly. If you'd like, I can just make a new pull request so that everything is in one commit.

File: README.md [hunk 4]
```diff
@@ -18,6 +18,48 @@ To get updates, watch the repo and follow the [Twitter account](https://twitter
 
 # Tips list
 
+## #11 - [Hoisting](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var#var_hoisting)
+> 2016-01-11 by [@squizzleflip](https://twitter.com/squizzleflip)
+
+Understanding hoisting will help you organize your function scope. Just remember, variable declaration and function definition are hoisted to the top. Variable definition is not, even if you declare and define a variable on the same line. Also, variable **declaration** is letting the system know that the variable exists while **definition** is assigning it a value.
```
Comment (loverajoel) [inline README.md]: Please can you remove the link from the title and add here to the second word hoisting?

"Understanding [hoisting](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var#var_hoisting) will help you organize..."

**Comment by loverajoel** [comment]
@seanpat09 don't worry about commits, I can do the squash for you after the merge!

**Comment by loverajoel** [comment]
@seanpat09 perfect, thanks!

**Pull request merged**